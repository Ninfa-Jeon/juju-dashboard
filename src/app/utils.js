import React from "react";

export const generateStatusIcon = (status, count) => {
  let statusClass = status ? `is-${status.toLowerCase()}` : "";
  let countValue = "";
  if (count !== undefined) {
    countValue = ` (${count})`;
  }
  return (
    <span className={"status-icon " + statusClass}>
      {status}
      {countValue}
    </span>
  );
};

export const generateSpanClass = (className, value) => {
  return <span className={className}>{value}</span>;
};
