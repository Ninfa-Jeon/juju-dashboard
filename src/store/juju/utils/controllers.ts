import { checkForJujuUpdate } from "@canonical/jujulib";
import { AdditionalController, Controller } from "../types";

/**
 * Returns the number of controllers that have an update available.
 * @param controllers The list of controllers to check for updates.
 * @returns
 */
export const countUpdates = (
  controllers?: (Controller | AdditionalController)[]
) => {
  let count = 0;
  if (controllers) {
    controllers.forEach((controller) => {
      if (
        "version" in controller &&
        controller.version &&
        checkForJujuUpdate(controller.version)
      )
        count += 1;
    });
  }
  return count;
};
