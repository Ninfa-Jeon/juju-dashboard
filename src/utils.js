/**
 * splitStr
 * Split strings on a given character
 *
 * @param {string} str - A string param
 * @param {string} character on which to split string
 * @return {array} Array of strings split
 */
export function splitStr(str, delimiter) {
  return str.split(delimiter);
}
