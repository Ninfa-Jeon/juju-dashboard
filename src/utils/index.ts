export { argPath } from "./argPath";
export { default as getMajorMinorVersion } from "./getMajorMinorVersion";
export { default as getUserName } from "./getUserName";
