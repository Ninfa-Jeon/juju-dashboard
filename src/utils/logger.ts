import log from "loglevel";

declare global {
  interface Window {
    logger: log.Logger;
  }
}

window.logger = log.getLogger("JujuDashboard");

export const logger = log.getLogger("JujuDashboard");
