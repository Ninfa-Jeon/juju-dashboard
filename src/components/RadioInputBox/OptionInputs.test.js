import { mount } from "enzyme";
import { Formik } from "formik";
import { waitForComponentToPaint } from "testing/utils";

import OptionInputs from "./OptionInputs";

describe("OptionInputs", () => {
  const options = [
    {
      name: "fruit",
      description: "yum",
      type: "string",
      required: true,
    },
    {
      name: "veg",
      description: "ehhh",
      type: "string",
      required: false,
    },
  ];

  it("renders the supplied list of options", () => {
    const onValuesChange = jest.fn();
    const wrapper = mount(
      <Formik>
        <OptionInputs
          name="MyAction"
          options={options}
          onValuesChange={onValuesChange}
        />
      </Formik>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("On input value change calls the onValuesChange handler", async () => {
    const onValuesChange = jest.fn();
    const wrapper = mount(
      <Formik
        initialValues={{
          "MyAction-fruit": "",
          "MyAction-veg": "",
        }}
      >
        <OptionInputs
          name="MyAction"
          options={options}
          onValuesChange={onValuesChange}
        />
      </Formik>
    );
    await waitForComponentToPaint(wrapper);
    // Formik fires a change event for the initialValues set.
    expect(onValuesChange).toHaveBeenCalledTimes(1);

    wrapper.find('input[name="MyAction-fruit"]').simulate("change", {
      target: {
        name: "MyAction-fruit",
        value: "foo",
      },
    });

    await waitForComponentToPaint(wrapper);
    expect(onValuesChange).toHaveBeenCalledTimes(2);
    expect(onValuesChange.mock.calls[1]).toEqual([
      "MyAction",
      { "MyAction-fruit": "foo", "MyAction-veg": "" },
    ]);
  });
});
