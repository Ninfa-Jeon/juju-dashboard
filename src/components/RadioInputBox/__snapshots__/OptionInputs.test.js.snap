// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionInputs renders the supplied list of options 1`] = `
<Formik>
  <OptionInputs
    name="MyAction"
    onValuesChange={
      [MockFunction] {
        "calls": Array [
          Array [
            "MyAction",
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    options={
      Array [
        Object {
          "description": "yum",
          "name": "fruit",
          "required": true,
          "type": "string",
        },
        Object {
          "description": "ehhh",
          "name": "veg",
          "required": false,
          "type": "string",
        },
      ]
    }
  >
    <form>
      <div
        className="radio-input-box__input-group"
        key="fruitInputGroup"
      >
        <label
          className="radio-input-box__label is-required"
          htmlFor="MyAction-fruit"
        >
          fruit
        </label>
        <Field
          className="radio-input-box__input"
          id="MyAction-fruit"
          name="MyAction-fruit"
          type="text"
        >
          <input
            className="radio-input-box__input"
            id="MyAction-fruit"
            name="MyAction-fruit"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          />
        </Field>
        <DescriptionSummary
          description="yum"
        >
          <span>
            yum
          </span>
        </DescriptionSummary>
      </div>
      <div
        className="radio-input-box__input-group"
        key="vegInputGroup"
      >
        <label
          className="radio-input-box__label"
          htmlFor="MyAction-veg"
        >
          veg
        </label>
        <Field
          className="radio-input-box__input"
          id="MyAction-veg"
          name="MyAction-veg"
          type="text"
        >
          <input
            className="radio-input-box__input"
            id="MyAction-veg"
            name="MyAction-veg"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
          />
        </Field>
        <DescriptionSummary
          description="ehhh"
        >
          <span>
            ehhh
          </span>
        </DescriptionSummary>
      </div>
    </form>
  </OptionInputs>
</Formik>
`;
